### PR TITLE
[18066] Monitor Service Documentation

### PR DESCRIPTION
**Pull request by Mario-DL**
This PR adds the proper documentation for

* eProsima/Fast-DDS#3894

This PR should be merged after the `v2.12.x` brach out is done!

### CODE REVIEW

File: docs/fastdds/monitor_service/monitor_service.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.. include:: ../../03-exports/aliases.include
+.. include:: ../../03-exports/aliases-api.include
+
+.. _monitor_service:
+
+Monitor Service
+===============
+
+The *Fast DDS* Monitor Service is an feature of Fast DDS that
```
Comment (jsantiago-eProsima) [inline docs/fastdds/monitor_service/monitor_service.rst:9]:
```suggestion
The *Fast DDS* Monitor Service is a feature of Fast DDS that
```

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: code/DDSCodeTester.cpp [hunk 2]
```diff
@@ -646,6 +646,34 @@ void dds_domain_examples()
...
+        // FASTDDS_MONITOR_SERVICE_API
+
+        DomainParticipant* participant_with_mon_srv = DomainParticipantFactory::get_instance()->create_participant(0,
+                        PARTICIPANT_QOS_DEFAULT);
+
+        // Enable Fast DDS Monitor Service through API
+        participant_with_mon_srv->enable_monitor_service();
+
+        // Disable Fast DDS Monitor Service through API
+        participant_with_mon_srv->enable_monitor_service();
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply code/DDSCodeTester.cpp:672]: Nice catch !

**Review by jsantiago-eProsima — APPROVED** [review]

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
Reviewed commit 340ab5d.

**Comment by JLBuenoLopez-eProsima** [comment]
Please, fix the CI failures: there is a line over the allowed character limit.

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM

File: docs/fastdds/statistics/monitor_service/intro.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+Keywords
+^^^^^^^^
+
+* **Proxy**: An entity that acts on behalf of another entity.
+
+* **Proxy Data**: The way in which a Proxy can be described.
+
+* **Monitoring Information**: The collection of different sources of information and statuses of an entity,
+  including: the Proxy Data, incompatible qos, connections, liveliness, deadlines missed, inconsistent
```
Comment (JesusPoderoso) [inline docs/fastdds/statistics/monitor_service/intro.rst:23]:
```suggestion
  including: the Proxy data, incompatible QoS, connections, liveliness, deadlines missed, inconsistent
```

File: docs/fastdds/statistics/monitor_service/monitor_service_configuration.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+| .. code-block:: bash                                     |
+|                                                          |
+|    export FASTDDS_STATISTICS="MONITOR_SERVICE_TOPIC"     |
++----------------------------------------------------------+
+| **Environment Variable Windows**                         |
++----------------------------------------------------------+
+| .. code-block:: bash                                     |
+|                                                          |
+|    set FASTDDS_STATISTICS=MONITOR_SERVICE_TOPIC          |
++----------------------------------------------------------+
```
Comment (JesusPoderoso) [inline docs/fastdds/statistics/monitor_service/monitor_service_configuration.rst:55]: Tabs?

```suggestion
.. tabs::

  .. group-tab:: C++ API

    .. code-block:: cpp

      .. literalinclude:: /../code/DDSCodeTester.cpp
        :language: c++
        :dedent: 8
        :start-after: // FASTDDS_MONITOR_SERVICE_API
        :end-before: //!

    .. group-tab:: C++ Property

      .. literalinclude:: /../code/DDSCodeTester.cpp
        :language: c++
        :dedent: 8
        :start-after: // FASTDDS_MONITOR_SERVICE_PROPERTY
        :end-before: //!

    .. group-tab:: XML

      .. literalinclude:: /../code/XMLTester.xml
        :language: xml
        :start-after: <!-->DDS_MONITOR_SERVICE<-->
        :end-before: <!--><-->

    .. group-tab:: Environment Variable Linux

      .. code-block:: bash

        export FASTDDS_STATISTICS="MONITOR_SERVICE_TOPIC"

    .. group-tab:: Environment Variable Windows

      .. code-block:: bash

        set FASTDDS_STATISTICS=MONITOR_SERVICE_TOPIC
```

File: docs/fastdds/statistics/monitor_service/intro.rst (hunk 3 above)
Comment (Mario-DL) [reply docs/fastdds/statistics/monitor_service/intro.rst:23]: (Applied the QoS thing)

File: code/DDSCodeTester.cpp [hunk 5]
```diff
@@ -647,6 +647,39 @@ void dds_domain_examples()
...
         //!--
     }
+    {
+        // FASTDDS_MONITOR_SERVICE_PROPERTY
+        DomainParticipantQos pqos;
+
+        // Activate Fast DDS Monitor Service through properties
+        pqos.properties().properties().emplace_back("fastdds.enable_monitor_service",
+                "true");
+
```
Comment (JesusPoderoso) [inline code/DDSCodeTester.cpp:657]: Include the other way to enable monitor service through the statistics

```suggestion
        // Activate Fast DDS Monitor Service through properties
        pqos.properties().properties().emplace_back("fastdds.enable_monitor_service",
                "true");
                
        // Activate Fast DDS Monitor Service through statistics properties (other way)
        pqos.properties().properties().emplace_back("fastdds.statistics",
                "MONITOR_SERVICE_TOPIC");

```

File: docs/fastdds/statistics/monitor_service/monitor_service_configuration.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        :start-after: // FASTDDS_MONITOR_SERVICE_PROPERTY
+        :end-before: //!
+
+    .. tab:: XML
+
+      .. literalinclude:: /../code/XMLTester.xml
+        :language: xml
+        :start-after: <!-->DDS_MONITOR_SERVICE<-->
+        :end-before: <!--><-->
+
```
Comment (JesusPoderoso) [inline docs/fastdds/statistics/monitor_service/monitor_service_configuration.rst:43]: Fix XML to display commented header section and append closing tag `</profiles>` at the end.

```suggestion
      .. literalinclude:: /../code/XMLTester.xml
        :language: xml
        :start-after: <!-->DDS_MONITOR_SERVICE<-->
        :end-before: <!--><-->
        :lines: 2-3,5-22
        :append: </profiles>

```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**